WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt

```diff
@@ -0,0 +1,172 @@
+    val url: String? = null,
+    val errorType: Int? = null
+) : WizardState, Parcelable
+
+typealias NavigationTarget = WizardNavigationTarget<RestoreStep, RestoreState>
+
+class RestoreViewModel @Inject constructor(
+    private val wizardManager: WizardManager<RestoreStep>
+) : ViewModel() {
+    private var isStarted = false
+
+    private lateinit var restoreState: RestoreState
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> by lazy {
+        SingleEventObservable(
+                Transformations.map(wizardManager.navigatorLiveData) {
+                    clearOldRestoreState(it)
+                    WizardNavigationTarget(it, restoreState)
+                }
+        )
+    }
+
+    private val _wizardFinishedObservable = MutableLiveData<Event<RestoreWizardState>>()
+    val wizardFinishedObservable: LiveData<Event<RestoreWizardState>> = _wizardFinishedObservable
+
+    private val _toolbarStateObservable = MutableLiveData<ToolbarState>()
+    val toolbarStateObservable: LiveData<ToolbarState> = _toolbarStateObservable
+
+    private val _snackbarEvents = MediatorLiveData<Event<SnackbarMessageHolder>>()
+    val snackbarEvents: LiveData<Event<SnackbarMessageHolder>> = _snackbarEvents
+
+    private val _errorEvents = MediatorLiveData<Event<RestoreErrorTypes>>()
+    val errorEvents: LiveData<Event<RestoreErrorTypes>> = _errorEvents
+
+    private val _navigationEvents = MediatorLiveData<Event<RestoreNavigationEvents>>()
+    val navigationEvents: LiveData<Event<RestoreNavigationEvents>> = _navigationEvents
+
+    fun start(savedInstanceState: Bundle?) {
+        if (isStarted) return
+        isStarted = true
+        if (savedInstanceState == null) {
+            restoreState = RestoreState()
+            // Show the next step only if it's a fresh activity so we can handle the navigation
+            wizardManager.showNextStep()
+        } else {
+            restoreState = requireNotNull(savedInstanceState.getParcelable(KEY_RESTORE_STATE))
+            val currentStepIndex = savedInstanceState.getInt(KEY_RESTORE_CURRENT_STEP)
+            wizardManager.setCurrentStepIndex(currentStepIndex)
+        }
+    }
+
+    fun addSnackbarMessageSource(snackbarEvents: LiveData<Event<SnackbarMessageHolder>>) {
+        _snackbarEvents.addSource(snackbarEvents) { event ->
+            _snackbarEvents.value = event
+        }
+    }
+
+    fun addErrorMessageSource(errorEvents: LiveData<Event<RestoreErrorTypes>>) {
+        _errorEvents.addSource(errorEvents) { event ->
+            _errorEvents.value = event
+        }
+    }
+
+    fun writeToBundle(outState: Bundle) {
+        outState.putInt(KEY_RESTORE_CURRENT_STEP, wizardManager.currentStep)
+        outState.putParcelable(KEY_RESTORE_STATE, restoreState)
+    }
+
+    fun onBackPressed() {
+        when (wizardManager.currentStep) {
+            DETAILS.id -> {
+                _wizardFinishedObservable.value = Event(RestoreCanceled)
+            }
```

Comment:
Question (❓): Why not adding the `PROGRESS.id` and `COMPLETE.id` cases to match that with the `BackupDownload` equivalent view model logic? Especially since the `RestoreInProgress` and `RestoreCompleted` objects are already there? Will that logic change for the `Restore` screen?

Reply:
They will be added as I move onto those steps - tried to keep this PR focused to just enough so that restore is shown